src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -64,25 +84,56 @@ export const QueryLanguageSelector = (props: Props) => {
       )?.label as string) ?? languageOptions[0].label,
   };
 
-  const handleLanguageChange = (newLanguage: EuiComboBoxOptionOption[]) => {
-    const queryLanguage = newLanguage[0].value as string;
-    props.onSelectLanguage(queryLanguage);
-    uiService.Settings.setUserQueryLanguage(queryLanguage);
+  const handleLanguageChange = (newLanguage: any) => {
+    props.onSelectLanguage(newLanguage);
+    uiService.Settings.setUserQueryLanguage(newLanguage);
   };
 
-  uiService.Settings.setUserQueryLanguage(props.language);
+  // language selector at the footer of the query bar
+  const button = (
+    <EuiButtonEmpty
+      iconSide="right"
+      onClick={onButtonClick}
+      style={{
+        fontSize: 'small',
+        height: '20px',
+      }}
+    >
+      {selectedLanguage.label}
+    </EuiButtonEmpty>
+  );
+  const getIconType = (language: string) => {
+    return language === selectedLanguage.label ? 'check' : 'empty';
+  };
 
+  const languageOptionsMenu = languageOptions.map((language) => {
+    return (
+      <EuiContextMenuItem
+        key={language.label}
+        icon={getIconType(language.label)}
```

Comment:
can maybe do a 1 line function for this instead of making another function where we checked it here